src/common.rs

```diff
@@ -1,7 +1,59 @@
 //! Common types and definitions.
 
+use bitflags::bitflags;
+
 /// Thread ID
 pub type Tid = core::num::NonZeroUsize;
 
 /// Process ID
 pub type Pid = core::num::NonZeroUsize;
+
+bitflags! {
+    // The read/write flags below may look a little weird, but that is the way
+    // they are defined in the protocol.
+    /// Host flags for opening files.
```

Comment:
please include a link back to the relevant GDB docs that define these constants as part of the doc comment.

ditto for `HostMode`